verifier/verifier.go

```diff
@@ -18,16 +18,7 @@ func Validate(proof shared.MerkleProof, labelHashFunc func(data []byte) []byte,
 ) error {
 	if int(securityParam) != len(proof.ProvenLeaves) {
 		return fmt.Errorf("number of proven leaves (%d) must be equal to security param (%d)",
-			len(proof.ProvenLeaves), securityParam)
-	}
-	if len(proof.ProvenLeaves)*36 < len(proof.ProofNodes) {
-		return fmt.Errorf("for every proven leaf (%d) there must be at most 36 proof nodes (%d)",
-			len(proof.ProvenLeaves), len(proof.ProofNodes),
-		)
-	}
```

Comment:
This check should be fine, shouldn't it? The same is enforced in scale limits.

Reply:
Scale limits enforce that no more than 5400 nodes are contained in a proof and 150 leaves. I'm not sure if > 36 nodes can exist per leave, but there definitely can be less than 1. 

With both checks in place all unit tests pass, except for mac which sometimes fails because there are fewer nodes than leaves. System tests also fail with the checks in place.

I didn't want to assume anything about how the trees are constructed if this change already lead to random failures and instead only keep the check that avoids a panic. 🙂

Reply:
There can be fewer nodes per leaf because for some leaves the paths from a leaf to the root of the tree can overlap.